ExampleMod/Common/GlobalProjectiles/ProjectileWithGrowingDamage.cs

```diff
@@ -0,0 +1,41 @@
+﻿using ExampleMod.Common.GlobalItems;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Terraria;
+using Terraria.DataStructures;
+using Terraria.ModLoader;
+
+namespace ExampleMod.Common.GlobalProjectiles
+{
+	public class ProjectileWithGrowingDamage : GlobalProjectile
+	{
+		//Related to GlobalItem: WeaponWithGrowingDamage
+
+		private Item sourceItem;
+
+		public override bool InstancePerEntity => true;
+		public override void OnSpawn(Projectile projectile, IEntitySource source) {
+			if (source is EntitySource_ItemUse itemSource) {
+				sourceItem = itemSource.Item;
+			}
+			else if (source is EntitySource_ItemUse_WithAmmo itemWithUseAmmoSource) {
+				sourceItem = itemWithUseAmmoSource.Item;
+			}
+		}
+
+		public override void OnHitNPC(Projectile projectile, NPC target, int damage, float knockback, bool crit) {
+			if (sourceItem == null || !sourceItem.TryGetGlobalItem(out WeaponWithGrowingDamage weapon))
```

Comment:
The best you can do instead, is to check `plr.inventory[plr.selectedItem]`. I'd suggest saving the `type` of item that spawned the projectile, and checking that too.

Reply:
Is that the same as player.HeldItem?  You could check it that way, but there would be no way for minion weapons to ever gain xp unless you held your minion staff the whole time.  I would make a list on the GlobalItem that tracks all projectiles spawned by the associated item instead of a projectile tracking the source item.

Reply:
`public Item HeldItem => inventory[selectedItem];`

Apparently, `HeldItem` is definitely good then. If you wanted this to work for minion items and such, then inventory search is definitely the way to go.

Actually... on second thoughts, you can't cache the `Item` reference, but you definitely can cache the `WeaponWithGrowingDamage` instance!

Reply:
Just make sure you comment in `OnSpawn` about the trap

Reply:
done